### PR TITLE
qemu.tests: Add case for nvdimm backed by host emulated device

### PR DESCRIPTION
**Pull request by huangyum**
Add case for nvdimm backed by host emulated device
ID: 1400771
Signed-off-by: Yumei Huang <yuhuang@redhat.com>

### CODE REVIEW

File: qemu/tests/cfg/nvdimm.cfg [hunk 1]
```diff
@@ -27,3 +27,12 @@
...
     variants:
         - nvdimm_basic:
+
+        - nvdimm_emulate:
+            # Please update host kernel line first, such as add "memmap=1G!5G": 
+            # 1G is the size, 5G is the start address of host memory
+            start_vm = "no"
+            nv_backend = "/dev/pmem0"
+            mem-path = ${nv_backend}
+            pre_command =
```
Comment (xutian) [inline qemu/tests/cfg/nvdimm.cfg]: use
```
del pre_command =
del post_command = 
```

File: qemu/tests/nvdimm.py [hunk 2]
```diff
@@ -1,7 +1,10 @@
 import logging
+import os
```
Comment (xutian) [inline qemu/tests/nvdimm.py]: take care  module import os please !!!

File: qemu/tests/nvdimm.py [hunk 3]
```diff
@@ -101,6 +104,15 @@ def run(test, params, env):
     :param params: Dictionary with the test parameters
     :param env: Dictionary with test environment.
     """
+    if params["start_vm"] == "no":
+        error.context("Check nvdimm backend in host", logging.info)
+        if os.path.exists(params["nv_backend"]):
```
Comment (xutian) [inline qemu/tests/nvdimm.py]: better,  
```

if not os.path.exists(params["nv_backend"]):
    raise error.TestError("nvdimm backend file ('%s')not exists in host " % param["nv_backend"])
params["start_vm"] = "yes"
env_process.preprocess_vm(test, params, env, params["main_vm"])
```

File: qemu/tests/nvdimm.py (hunk 2 above)
Comment (huangyum) [inline qemu/tests/nvdimm.py]: Could you please explain why?
Other parts are updated, please help review again, thanks!

File: qemu/tests/cfg/nvdimm.cfg [hunk 4]
```diff
@@ -27,3 +27,12 @@
...
         - nvdimm_basic:
+
+        - nvdimm_emulate:
+            # Please update host kernel line first, such as add "memmap=1G!5G": 
+            # 1G is the size, 5G is the start address of host memory
+            start_vm = "no"
+            nv_backend = "/dev/pmem0"
+            mem-path = ${nv_backend}
+            del pre_command
+            del post_command
```
Comment (luckyh) [inline qemu/tests/cfg/nvdimm.cfg]: It looks the feature is only supported on pc (and maybe q35), can you please create a restriction?

Comment (huangyum) [inline qemu/tests/cfg/nvdimm.cfg]: Thanks, updated!

File: qemu/tests/cfg/nvdimm.cfg [hunk 5]
```diff
@@ -27,3 +28,12 @@
 
     variants:
         - nvdimm_basic:
+
```
Comment (vivianQizhu) [inline qemu/tests/cfg/nvdimm.cfg]: Why there is a blank?

Comment (huangyum) [inline qemu/tests/cfg/nvdimm.cfg]: Updated, please review again, thanks!

**Comment by vivianQizhu** [comment]
# git am 876.patch
Applying: qemu.tests: Add case for nvdimm backed by host emulated device
/home/test/tp-qemu/.git/rebase-apply/patch:23: trailing whitespace.
            # Please update host kernel line first, such as add "memmap=1G!5G": 
warning: 1 line adds whitespace errors.

**Comment by huangyum** [comment]
@vivianQizhu Thanks, updated.  Seems inspekt didn't check out that. Please help review again, thanks!

File: qemu/tests/cfg/nvdimm.cfg [hunk 6]
```diff
@@ -27,3 +28,11 @@
 
     variants:
         - nvdimm_basic:
+        - nvdimm_emulate:
+            # Please update host kernel line first, such as add "memmap=1G!5G":
```
Comment (xutian) [inline qemu/tests/cfg/nvdimm.cfg]: Please add this check in your test, like
try:
     process.system(grep 'memmap='  /proc/cmdline)
except process.CmdError:
     raise exceptions.TestError("Please add kernel param 'memmap' before start test.")

**Comment by huangyum** [comment]
@xutian updated, please help review again, thanks!

**Comment by xutian** [comment]
looks OK to me.

**Comment by xutian** [comment]
Merged, thanks @huangyum and @vivianQizhu

**Pull request merged**